hydroflow/examples/kvs/server.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{GraphType, Opts};
+
+use crate::helpers::{
+    deserialize_msg, is_get, is_put, resolve_ipv4_connection_addr, serialize_msg,
+};
+use crate::protocol::KVSMessage;
+
+use hydroflow::hydroflow_syntax;
+use hydroflow::scheduled::graph::Hydroflow;
+use std::net::SocketAddr;
+use tokio::net::UdpSocket;
+
+pub(crate) async fn run_server(opts: Opts) {
+    // First, set up the socket
+
+    let server_addr = resolve_ipv4_connection_addr(opts.addr, opts.port)
+        .expect("Unable to bind to provided IP and port");
+    let server_socket = UdpSocket::bind(server_addr).await.unwrap();
+    let (outbound, inbound) = hydroflow::util::udp_lines(server_socket);
+    println!("Listening on {}", server_addr);
+    println!("Server live!");
+
+    let mut df: Hydroflow = hydroflow_syntax! {
+        // NW channels
+        outbound_chan = merge() -> map(|(m,a)| (serialize_msg(m), a)) -> sink_async(outbound);
+        inbound_chan = recv_stream(inbound) -> map(deserialize_msg) -> tee();
+        puts = inbound_chan[0] -> filter_map(is_put) -> tee();
+        gets = inbound_chan[1] -> filter_map(is_get) -> tee();
```

Comment:
Will be able to roll this (and optionally a bit of the boilerplate operators below) into demux when thats merged, up to you if you want to merge this first or wait